MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -49,6 +52,37 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
         executeJavaScriptCallback(responseStatus: .onSuccess, messageId: messageId, response: uniqueId)
     }
 
+    func requestPermission(requestParam: RequestParameters?, callbackId: String) {
+        guard let requestParamValue = requestParam?.permission else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: MiniAppJavaScriptError.invalidPermissionType.rawValue)
+            return
+        }
+        guard let requestPermissionType = MiniAppPermissionType(rawValue: requestParamValue) else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: MiniAppJavaScriptError.invalidPermissionType.rawValue)
+            return
+        }
+
+        switch requestPermissionType {
+        case .location:
+            getPermissionResult(requestPermissionType: requestPermissionType, callbackId: callbackId)
+        }
+    }
+
+    func getPermissionResult(requestPermissionType: MiniAppPermissionType, callbackId: String) {
+        hostAppMessageDelegate?.requestPermission(permissionType: requestPermissionType) { (result) in
+            switch result {
+            case .success:
+                self.executeJavaScriptCallback(responseStatus: .onSuccess, messageId: callbackId, response: "Allowed")
+            case .failure(let error):
+                if !error.localizedDescription.isEmpty {
+                    self.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: error.localizedDescription)
```

Comment:
I see that if users deny permission then it will execute error callback on js bridge.
I guess android should do the same? @corycaywood

Reply:
Yeah, it's probably better to use the error callback when denied permission.